big-o/binary-search.md

```diff
@@ -39,8 +39,8 @@ def binary_search(array, value):
         else:
             return mid
 
-        if array[low] == value:
-            return low
+    if array[low] == value:
+        return low
```

Comment:
Despite the detail in the asana task, I don't think this fies the implementation. This check has always felt redundant to me. The code also fails if the array is empty, or if the value comes after the values that are present. After adjusting for those cases, I ran this code against another implementation lacking this check over a wide combination of lists ranging in size from 0 to 20 elements, and got matching results for the following implementation that I find to be cleaner:
```py
def binary_search(array, value):
    low = 0
    high = len(array)
    while low < high:
        mid = (low + high)//2
        if array[mid] > value:
            high = mid
        elif array[mid] < value:
            low = mid + 1
        else:
            return mid

    return None
```

Not only does it avoid that extra check (which seems unnecessary in the end), but it also uses the principle of the top end of a range (high) being exclusive, just like any other python range.

I would also suggest updating the first challenge on this page to reflect the updated structure as well (some of the names differ).